app/src/main/java/com/example/trempel/ui/pdp/PdpFragment.kt

```diff
@@ -32,14 +37,35 @@ internal class PdpFragment : Fragment() {
             this.lifecycleOwner = this@PdpFragment
             this.viewModel = this@PdpFragment.viewModel
         }.also {
+            it.initializationRecyclerAdapter()
             viewModel.loadProduct(args.productId)
             observeExceptionResponse()
+            viewModel.getRecentlyViewedProduct(args.productId)
+            observeRecentlyViewedProduct()
```

Comment:
we should start observing the data before the actual call

Reply:
done